service.subtitles.rvm.addic7ed/addic7ed/core.py

```diff
@@ -0,0 +1,253 @@
+# coding: utf-8
+# Created on: 07.04.2016
+# Author: Roman Miroshnychenko aka Roman V.M. (roman1972@gmail.com)
+
+from __future__ import absolute_import, unicode_literals
+from future import standard_library
+standard_library.install_aliases()
+from builtins import str, dict
+
+import os
+import sys
+import re
+import shutil
+from collections import namedtuple
+from urllib import parse as urlparse
+from kodi_six import xbmc, xbmcplugin, xbmcgui, xbmcvfs
+from . import parser
+from .addon import addon, profile, get_ui_string, icon
+from .exceptions import DailyLimitError, ParseError, SubsSearchError
+from .utils import logger, get_languages, get_now_played, parse_filename, \
+    normalize_showname
+
+__all__ = ['router']
+
+temp_dir = os.path.join(profile, 'temp')
+handle = int(sys.argv[1])
+
+
+VIDEOFILES = ('.avi', '.mkv', '.mp4', '.ts', '.m2ts', '.mov')
+dialog = xbmcgui.Dialog()
+release_re = re.compile(r'-(.*?)(?:\[.*?\])?\.')
+
+EpisodeData = namedtuple('EpisodeData',
+                         ['showname', 'season', 'episode', 'filename'])
+
+
+def display_subs(subs_list, episode_url, filename):
+    """
+    Display the list of found subtitles
+
+    :param subs_list: the list of named tuples with the following fields:
+
+        - language: Kodi language name for the subtitles.
+        - verison: a descriptive text for the subtitles.
+        - hi (bool): ``True`` if subs for hearing impaired
+        - link: download link for the subtitles.
+
+    :param episode_url: the URL for the episode page on addic7ed.com.
+        It is needed for downloading subs as 'Referer' HTTP header.
+    :param filename: the name of the video-file being played.
+
+    Each item in the displayed list is a ListItem instance with the following
+    properties:
+
+    - label: Kodi language name (e.g. 'English')
+    - label2: a descriptive text for subs
+    - thumbnailImage: a 2-letter language code (e.g. 'en') to display a country
+      flag.
+    - 'hearing_imp': if 'true' then 'CC' icon is displayed for the list item.
+    - 'sync': if 'true' then 'SYNC' icon is displayed for the list item.
+    - url: a plugin call URL for downloading selected subs.
+    """
+    for item in subs_list:
+        list_item = xbmcgui.ListItem(
+            label=item.language, label2=item.version,
+            thumbnailImage=xbmc.convertLanguage(item.language, xbmc.ISO_639_1)
```

Comment:
please use `list_item.setArt()` since `thumbnailImage` is deprecated

Reply:
I'm not sure if this rule applies to subtitles API, because `thumbnailImage` here is not an actual image, but a language code, like `'en'` or `'uk'`.

Reply:
UPD: `setArt()` seems to work too. I'll do some more test and update the PR.